habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDepsMojo.java

```diff
@@ -264,4 +288,31 @@ protected void setAnchorSourceDirectory(File anchorSourceDirectory) {
     public File getAnchorOutputDirectory() {
         return anchorOutputDirectory;
     }
+
+    protected void setDockerfile(File dockerfile) {
+        this.dockerfile = dockerfile;
+    }
+
+    protected void performDockerfileUpdateForVirtualEnvironment() {
+        String updatedDockerfile =
+                DockerfileUtil.updateDockerfileWithContainerStageLogic(dockerfile,
+                DockerfileUtil.getContainerStageContentFrom(getBuilderStageTemplateFileName(), this.anchorSourceDirectory.getName(), this.singleHabushuDependencyBaseDir),
+                DockerfileUtil.getContainerStageContentFrom(getFinalStageTemplateFileName(), null, null));
+
+        try (Writer writer = new FileWriter(dockerfile)) {
+            writer.write(updatedDockerfile);
+
+        } catch (IOException e) {
+            throw new HabushuException("Unable to update Dockerfile.", e);
+        }
+    }
+
+    protected String getBuilderStageTemplateFileName() {
+        return "dockerfile_builder_stage_template";
+    }
+
+    protected String getFinalStageTemplateFileName() {
+        return "dockerfile_final_stage_template";
+    }
```

Comment:
S: these should just be final strings like `protected final String FINAL_STAGE_TEMPLATE = "dockerfile_final_stage_template"`